app/models/inning.rb

```diff
@@ -9,13 +11,41 @@ class Inning < ActiveRecord::Base
   validates :half, presence: true, uniqueness: { scope: [:game, :number] }
   validates :status, presence: true
 
+  validates_inclusion_of :outs, :in => 0..3
+
   validates :number, numericality: { greater_than_or_equal_to: 1 }
   validates :runs, numericality: { greater_than_or_equal_to: 0 }
 
+  after_initialize :set_default_values
   before_save :ensure_inning_sequence
+  before_save :update_status, unless: :destroyed?
+  after_save :update_game_score, unless: :destroyed? #TODO: switch back to after_commit (?)
```

Comment:
Added. Keeping the callback as `after_save` for now to avoid scope creep as we'll have to add in a new gem to test after_commit callbacks.